site/app/repositories/email/EmailRepository.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace app\repositories\email;
+
+use app\entities\email\EmailEntity;
+use app\models\EmailStatusModel;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Tools\Pagination\Paginator;
+
+class EmailRepository extends EntityRepository {
+    const PAGE_SIZE = 1000;
+    const MAX_SUBJECTS_PER_PAGE = 10;
+
+    public function getEmailsByPage(int $page, $semester = null, $course = null): Paginator {
+        $entity = EmailEntity::class;
+        $course_specific = ($semester && $course) ? "WHERE e.semester = '{$semester}' AND e.course = '{$course}' " : "";
+        $dql = "SELECT e FROM {$entity} e {$course_specific} ORDER BY e.created DESC";
```

Comment:
You should do something like :course_specific and then when you create the query use the setParamter function on course_specific. A good example is in #6708.